### ISSUE TITLE
Settings in libertine apps are not persistent

### ISSUE THREAD

**Issue by mymike00** [ubports/ubuntu-touch#812]
- Device: bq e4.5 krillin
- Channel: rc
- Build: 2018-w31

### Steps to reproduce

change the settings in a libertine app (I tried with aisle-riot, showing some toolbars), close and re-open the app

### Expected behavior

the settings I changed should be persistent (I expect to see the toolbars I set to visible)

### Actual behavior

the settings loaded are the default one, ignoring my changes before closing the app

### Logfiles and additional information

should I add logfiles? which ones?

**Comment by doniks** [comment]
did you close the app from the "in-app" menu? You know something like, File->Quit or similar? Or did you swipe it down?

If you quit, then this might be an application specific problem and might not at all be related to UT. In this case please adjust the issue title, because then we are talking about a specific application. Also please test whether that same bug exists in "normal" desktop ubuntu. If that is also the case, then really, this issue should be in that applications bugtracker not in UT. 

If you swiped down, then my suspicion is that UT is "killing" the app too quickly/aggressively. Maybe there can be some improvements like sending some kind of "please close" message from UT to the application, so the application has time to store it's settings etc. How does gnome/kde/other DEs close an application.

**Comment by mymike00** [comment]
Yeah, forgot to specify... Close with the swipe, as a normal ut app
Idk if the problem is while closing, shouldn't the app store the settings when I change them, and not when I close the app?

**Comment by Flohack74** [comment]
A libertine app does not expect to be killed so it saves nothing I assume. Please always use the application-specific way to terminate it. Its like you kill it from task manager ;)

**Comment by mymike00** [comment]
Using the in-app close button is hard, menus are unusable...

**Comment by Flohack74** [comment]
Thats what we always say, its a phone and not a desktop ;)

**Comment by doniks** [comment]
@Flohack74 , don't you think there might be a relatively simply improvement possible here? I mean on my desktop I also don't always close an application from the menu. Maybe I right click-close from some taskbar, or tap the x in this app-thumbnail-overview in gnome (whats it called?).

It seems to me we just terminate the app a bit too quickly for it to have a chance to clean up. Maybe it's as simple as, SIGTERM before you SIGKILL (I doubt that UT does that, but who knows), or maybe it's a matter of of SIGCONT the frozen background app before you SIGTERM (I also doubt that this is the actual problem, because I think libertine apps aren't really frozen in the background or are they?, or maybe maybe we just need to SIGTERM all the processes running in this XMir before we SIGTERM the X server itself

can anyone point to the "swipe down" code?

**Comment by mymike00** [comment]
> Thats what we always say, its a phone and not a desktop ;)
So why are we doing all these libertine stuffs? Just for the nexus 5 with slimport cable? But also in that case closing libertine apps from in-app method isn't the best way IMHO... On the desktop I close apps with the red x at thecorner of the window, that is on the syatem side and not in-app way...

**Comment by mymike00** [comment]
I tried closing from the menu inside the app (aisle-riot) using a mouse but when I re-open the app, settings I applied are reset and loaded the default ones

**Comment by Flohack74** [comment]
I am out of the game as I do not know internals of the unity8 method of closing an app. But apparently there is a difference if you kill the App from a task manager or you use the window manager buttons. But if the in-app shutdown does not save the settings then it might probably trying to write to the wrong (read-only) path. Can you find out where the settings should go?

**Comment by mymike00** [comment]
I could, but I have no idea about how doing it...😊

**Comment by ShadowEO** [comment]
@Flohack74 is UT still getting Locally Integrated Menus under Mir and Xmir or has that idea been scrapped with the transition from Canonical? As that could possibly solve the problem of the menus being too small.

**Comment by Flohack74** [comment]
@ShadowEO Sry no idea what you are talking about ;)

**Comment by NeoTheThird** [comment]
@ShadowEO 

> @Flohack74 is UT still getting Locally Integrated Menus under Mir and Xmir or has that idea been scrapped with the transition from Canonical? As that could possibly solve the problem of the menus being too small.

Yes: https://github.com/ubports/unity8/issues/16

**Comment by mymike00** [comment]
I also noticed that after opening menus a lot of time and trying to select something from them, unity 8 rebooted

**Comment by dobey** [comment]
This has nothing to do with how the app is opened or closed. Just to clarify that.
